### PR TITLE
return promise if callback not passed to async methods

### PR DESCRIPTION
**Pull request by jstewmon**
This is based on #155 and #156 . It could use a few more tests, but I wanted to get some feedback before writing any more.

BTW, vows seems dead, and has terrible ergonomics for async tests. Any interest in switching to something else? I'm partial to ava these days, but literally anything would be better than vows. :-)

### CODE REVIEW

**Comment by stash** [comment]
@jstewmon regarding vows: yeah, that seemed like a good idea at the time. I agree literally anything is better than vows ;)

**Comment by medelibero-sfdc** [comment]
Hi @jstewmon,

  Any chance you could revise the latest in main? Also, we will get this merged but first we want to get #156 merged and then this one. 

  Thanks,
  Mike

**Comment by jstewmon** [comment]
@medelibero-sfdc 👋  , I'm working on rebasing #156 now. Let's please make this the last time I have have to rebase - #156 has so many line moves that it is very tedious to integrate long runs of changes from master. 🙏

**Comment by jstewmon** [comment]
@medelibero-sfdc unfortunately, #173 precludes a clean path forward here because the callback should always be the last argument...

Also, that PR did not update the interface in the Store base class.

I have addressed those two issues in #156 in a new commit that notes the breaking change.

**Comment by awaterma** [comment]
Thank you for your help with this @jstewmon; sorry to ask for so many rebases. We should be good going forward.

**Comment by jstewmon** [comment]
@awaterma it wasn't too much trouble until other PRs started being accepted while #156 lingered... That PR is just especially difficult for merge tools to auto merge, so I have to go through everything by hand and fix up conflicts.

File: lib/memstore.js [hunk 1]
```diff
@@ -28,154 +28,160 @@
...
-// available.
-if (util.inspect.custom) {
-  MemoryCookieStore.prototype[util.inspect.custom] = MemoryCookieStore.prototype.inspect;
-}
-
-MemoryCookieStore.prototype.findCookie = function(domain, path, key, cb) {
-  if (!this.idx[domain]) {
-    return cb(null,undefined);
+"use strict";
+const { fromCallback } = require("universalify");
```
[the remark this answers is not in the document]
Comment (jstewmon) [reply lib/memstore.js]: [Destructuring assignment](https://node.green/#ES2015-syntax-destructuring--assignment-with-objects) is supported in node 6+.

**Comment by jstewmon** [comment]
@awaterma thanks for the review. I rebased on master and addressed your feedback.

**Comment by jstewmon** [comment]
👋 @awaterma freshly rebased if you have time to take another look 🙏

**Comment by awaterma** [comment]
I think this looks good; I'd like to discuss a bit with the open source term before merging. I'd rather not add an additional dependency; but "universalify" does provide the right solution. It's more of an extra dependency trade off discussion. :) Anyway, we should have good progress on this soon.

**Comment by jstewmon** [comment]
@awaterma thanks for pushing this over the line! 🙏 🎉 

Sorry for not responding over the last week - just been exceptionally busy at work. 😅

**Comment by awaterma** [comment]
We're grateful for the help! Thank you for contributing!

**Pull request merged**